src/backend/web.rs

```diff
@@ -6,106 +6,45 @@ use crate::{
-unsafe impl Send for AdapterId {}
-unsafe impl Sync for AdapterId {}
-
-#[derive(Debug, Clone)]
-pub(crate) struct DeviceId(web_sys::GpuDevice);
-unsafe impl Send for DeviceId {}
-unsafe impl Sync for DeviceId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct QueueId(web_sys::GpuQueue);
-unsafe impl Send for QueueId {}
-unsafe impl Sync for QueueId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct ShaderModuleId(web_sys::GpuShaderModule);
-unsafe impl Send for ShaderModuleId {}
-unsafe impl Sync for ShaderModuleId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct BindGroupLayoutId(web_sys::GpuBindGroupLayout);
-unsafe impl Send for BindGroupLayoutId {}
-unsafe impl Sync for BindGroupLayoutId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct BindGroupId(web_sys::GpuBindGroup);
-unsafe impl Send for BindGroupId {}
-unsafe impl Sync for BindGroupId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct TextureViewId(web_sys::GpuTextureView);
-unsafe impl Send for TextureViewId {}
-unsafe impl Sync for TextureViewId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct SamplerId(web_sys::GpuSampler);
-unsafe impl Send for SamplerId {}
-unsafe impl Sync for SamplerId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct BufferId(web_sys::GpuBuffer);
-unsafe impl Send for BufferId {}
-unsafe impl Sync for BufferId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct PipelineLayoutId(web_sys::GpuPipelineLayout);
-unsafe impl Send for PipelineLayoutId {}
-unsafe impl Sync for PipelineLayoutId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct TextureId(web_sys::GpuTexture);
-unsafe impl Send for TextureId {}
-unsafe impl Sync for TextureId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct RenderPipelineId(web_sys::GpuRenderPipeline);
-unsafe impl Send for RenderPipelineId {}
-unsafe impl Sync for RenderPipelineId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct ComputePipelineId(web_sys::GpuComputePipeline);
-unsafe impl Send for ComputePipelineId {}
-unsafe impl Sync for ComputePipelineId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct CommandEncoderId(web_sys::GpuCommandEncoder);
-unsafe impl Send for CommandEncoderId {}
-unsafe impl Sync for CommandEncoderId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct CommandBufferId(web_sys::GpuCommandBuffer);
-unsafe impl Send for CommandBufferId {}
-unsafe impl Sync for CommandBufferId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct SurfaceId(web_sys::GpuCanvasContext);
-unsafe impl Send for SurfaceId {}
-unsafe impl Sync for SurfaceId {}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct SwapChainId(web_sys::GpuSwapChain);
-unsafe impl Send for SwapChainId {}
-unsafe impl Sync for SwapChainId {}
+pub(crate) struct Sendable<T>(T);
+unsafe impl<T> Send for Sendable<T> {}
+unsafe impl<T> Sync for Sendable<T> {}
+
+pub(crate) type Context = Sendable<web_sys::Gpu>;
```

Comment:
Why/where do we need these typedefs?

Reply:
We don't, but you end up needing them in a lot of places, I can get rid of the typedefs (other than Context which is required) if you'd like.

Reply:
And certainly all of the type aliases other than `Context` don't need to be `pub(crate)`, that was a mistake.

Reply:
In the `impl Context` I think you can just use `Self::Buffer` instead of `Buffer` typedef, and same for everything else.
In the implementations for the passes we could just write plain `Sendable<websys::Buffer>`.
Would you be open for this? I think the less types internally defined, the better.

Reply:
Yeah that's completely fine I'll do that right now.